src/lazy/binary/raw/v1_1/type_code.rs

```diff
@@ -0,0 +1,147 @@
+use crate::IonResult;
+use std::convert::TryFrom;
+
+use crate::result::{IonError, IonFailure};
+use crate::IonType;
+
+/// Represents the type information found in the header byte of each binary Ion value.
+/// While this value can be readily mapped to a user-level [`IonType`], it is a distinct concept.
+/// The IonTypeCode enum captures system-level information that is not exposed to end users of the
+/// library, including:
+/// * Whether the cursor is positioned over whitespace that needs to be skipped.
+/// * Whether the integer value being read is positive or negative.
+/// * Whether the next type code is reserved.
+///
+/// See the
+/// [Typed Value Formats](https://amazon-ion.github.io/ion-docs/docs/binary.html#typed-value-formats)
+/// section of the spec for more information.
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+pub enum IonTypeCode {
+    EExpression(bool), // 0x00-0x4F - With, or without, address in the opcode.
```

Comment:
I think this would be more self-documenting as two variants; something like:
```suggestion
    EExpressionWithAddress, // 0x00-0x3F
    EExpressionAddressFollows // 0x40-0x4F
```

Of course, we're in the processing of changing the opcodes and the meaning of "address follows", so this can be a `TODO` for later.